IBM/ibm-common-service-operator
# Create isolate.sh

Currently designed for two runs. The first is to pause and the second is to restart the original cs instance
first called with `./isolate.sh --original-cs-ns <cs instance namespace> --control-ns <desired control namespace> --excluded-ns <comma separated list of namespaces to exclude from new common-service-maps>`
- gather parameters
- pause cs in original instance
- generate common-service-maps cm in kube-public

second `./isolate.sh --original-cs-ns <cs instance namespace> -r`
- check common-service-maps cm
- uninstall singletons
- restart original cs instance

Testing instructions:
- setup environment with preferably 3 cp namespaces sharing one cs (more cp namespaces, more combinations of excluded namespaces you can try)
- run isolate.sh script with varying combinations of excluded namespaces
    - between each run, if you delete the common-service-maps (after verifying it of course) and rescale ODLM and cs operator to 1, you can rerun the test "from scratch." This will let you test multiple combinations with minimal effort resetting the environment
- after verifying that the config map generated is correct, run the script with `-r` (shown above) and confirm pods start up as expected

Notes: 
- The generated csmaps will include the original cs namespace. This does not have any adverse impacts.
- if you run the restart after excluding some namespaces, pods in the excluded namespaces will be cutoff from common services. Expect those pods to fail or otherwise not function properly

Some outstanding changes to be made as of opening this pr:
- ensure ODLM sub `ISOLATED_MODE` value set to "true"
- error handling for excluded namespace not being set
- do not require control namespace to be set for `-r` option
- smaller tweaks based on feedback

Part of https://github.com/IBM/ibm-common-service-operator/pull/1099 moved to its own PR for clarity.
